components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/BasicClientAuthenticationHandler.java

```diff
@@ -0,0 +1,138 @@
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+
+import java.nio.charset.Charset;
+
+import static org.wso2.carbon.identity.auth.service.util.AuthConfigurationUtil.isAuthHeaderMatch;
+
+/**
+ * BasicClientAuthenticationHandler is for authenticate the request based on Basic Authentication
+ * using the client credentials.
+ * canHandle method will confirm whether this request can be handled by this authenticator or not.
+ */
+public class BasicClientAuthenticationHandler extends AuthenticationHandler {
+
+    private static final Log log = LogFactory.getLog(BasicClientAuthenticationHandler.class);
+    private final String BASIC_AUTH_HEADER = "Basic";
+
+    @Override
+    public void init(InitConfig initConfig) {
+
+    }
+
+    @Override
+    public String getName() {
+
+        return Constants.BASIC_CLIENT_AUTH_HANDLER;
+    }
+
+    @Override
+    public int getPriority(MessageContext messageContext) {
+
+        return getPriority(messageContext, 99);
+    }
+
+    @Override
+    public boolean canHandle(MessageContext messageContext) {
+
+        return isAuthHeaderMatch(messageContext, BASIC_AUTH_HEADER);
+    }
+
+    @Override
+    public boolean isEnabled(MessageContext messageContext) {
+
+        return true;
+    }
+
+    @Override
+    protected AuthenticationResult doAuthenticate(MessageContext messageContext) throws AuthenticationFailException {
+
+        AuthenticationResult authenticationResult = new AuthenticationResult(AuthenticationStatus.FAILED);
+        AuthenticationContext authenticationContext = (AuthenticationContext) messageContext;
+        String authorizationHeader = authenticationContext.getAuthenticationRequest().
+                getHeader(HttpHeaders.AUTHORIZATION);
+
+        if (authorizationHeader != null){
+            String[] splitAuthorizationHeader = authorizationHeader.split(" ");
+            if (splitAuthorizationHeader.length == 2) {
+                byte[] decodedAuthHeader = Base64.decodeBase64(splitAuthorizationHeader[1].getBytes());
+                String authHeader = new String(decodedAuthHeader, Charset.defaultCharset());
+                String[] credentials = authHeader.split(":", 2);
+
+                if (credentials.length == 2 && StringUtils.isNotBlank(credentials[0]) &&
+                        StringUtils.isNotBlank(credentials[1])) {
+                    String clientId = credentials[0];
+                    String clientSecret = credentials[1];
+
+                    try {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Authenticating application with client ID: " + clientId
+                                    + " with client secret.");
+                        }
+                        if (OAuth2Util.authenticateClient(clientId, clientSecret)) {
+                            authenticationResult.setAuthenticationStatus(AuthenticationStatus.SUCCESS);
+                        }
```

Comment:
else set AuthenticationStatus to false

Reply:
Addressed with ad728d29724eeaf1a63f72a7ab8f3f895f268c96